pkg/services/apiserver/storage/entity/utils.go

```diff
@@ -110,19 +110,12 @@ func resourceToEntity(res runtime.Object, requestInfo *request.RequestInfo, code
 	}
 	rv, _ := strconv.ParseInt(metaAccessor.GetResourceVersion(), 10, 64)
 
-	k := &entityStore.Key{
-		Group:       requestInfo.APIGroup,
-		Resource:    requestInfo.Resource,
-		Namespace:   requestInfo.Namespace,
-		Name:        metaAccessor.GetName(),
-		Subresource: requestInfo.Subresource,
-	}
+	k.Name = metaAccessor.GetName()
```

Comment:
should we clone the provided key rather than mutating it?

Reply:
good call. i will remove the pointer in the function arg to avoid mutation

Reply:
updated in https://github.com/grafana/grafana/commit/ec935986be833cf71819ae871d82f4d2e3c56a66